gears/src/test/java/org/hortonmachine/gears/modules/TestStacAsset.java

```diff
@@ -54,12 +55,35 @@ public void testCreateInvalidStacAssetNotACOG() throws JsonProcessingException {
 
     public void testCreateStacAssetRasterBandsMetadataMissingIsValid() throws JsonProcessingException {
         String assetJSON = "{\"title\":\"Band 1 (coastal) BOA reflectance\",\"type\":\"image/tiff; application=geotiff; profile=cloud-optimized\",\"roles\":[\"data\"],\"gsd\":60,\"eo:bands\":[{\"name\":\"B01\",\"common_name\":\"coastal\",\"center_wavelength\":0.4439,\"full_width_half_max\":0.027}],\"href\":\"https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/33/S/VB/2021/2/S2B_33SVB_20210221_0_L2A/B01.tif\",\"proj:shape\":[1830,1830],\"proj:transform\":[60,0,399960,0,-60,4200000,0,0,1],\"raster:bands\":[]}";
-        ObjectMapper mapper = new ObjectMapper();
         JsonNode node = mapper.readTree(assetJSON);
 
         HMStacAsset asset = new HMStacAsset("B01", node);
 
         assertTrue(asset.isValid());
     }
 
+    public void testReadRasterFromHTTP() throws Exception {
+        String assetJSON = "{\"title\":\"Band 1 (coastal) BOA reflectance\",\"type\":\"image/tiff; application=geotiff; profile=cloud-optimized\",\"roles\":[\"data\"],\"gsd\":60,\"eo:bands\":[{\"name\":\"B01\",\"common_name\":\"coastal\",\"center_wavelength\":0.4439,\"full_width_half_max\":0.027}],\"href\":\"https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/33/S/VB/2021/2/S2B_33SVB_20210221_0_L2A/B01.tif\",\"proj:shape\":[1830,1830],\"proj:transform\":[60,0,399960,0,-60,4200000,0,0,1],\"raster:bands\":[{\"data_type\":\"uint16\",\"spatial_resolution\":60,\"bits_per_sample\":15,\"nodata\":0,\"statistics\":{\"minimum\":1,\"maximum\":20567,\"mean\":2339.4759595597,\"stddev\":3026.6973619954,\"valid_percent\":99.83}}]}";
+        JsonNode node = mapper.readTree(assetJSON);
+        String assetId = "rainfall";
+        HMStacAsset hmAsset = new HMStacAsset(assetId, node);
+        RegionMap regionMap = RegionMap.fromBoundsAndGrid(1640000.0, 1640200.0, 5140000.0, 5140160.0, 20, 16);
+
+        GridCoverage2D grid = hmAsset.readRaster(regionMap);
+
+        assertNotNull(grid);
+    }
+
+    // need setup for AWS credentials in System
```

Comment:
@inigo-cobian would you mind adding here more information on how to setup said credentials? Maybe the necessary variables that one would need to set (if I understood correctly).

Reply:
I have found some example credentials in [the AWS documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-set) that we could use for testing. They can be defined as System properties or environment variables.

```java
System.setProperty("iio.s3.aws.region", "us-west-2");
System.setProperty("aws.accessKeyId", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY");
System.setProperty("aws.secretKey", "AKIAIOSFODNN7EXAMPLE");

```

```sh
export AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
export AWS_DEFAULT_REGION=us-west-2
```